### PR TITLE
V22F-293 update ipc lib

### PR DESCRIPTION
**Pull request by KungFuDonkey**
Updates the ipc lib to use initialization methods

### CODE REVIEW

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backendUnitTests/SocketBlackBoxTests.h [hunk 1]
```diff
@@ -24,21 +25,24 @@
...
 
 	/// intializes the black box with 2 tests
 	bb.Initialize(mock, situations, 2);
 	DecisionTuple decisions;
 
+	tCarElt car;
+	tSituation situation;
 	// no decision should be made yet
-	ASSERT_FALSE(bb.GetDecisions(mock, decisions));
+	ASSERT_FALSE(bb.GetDecisions(&car,&situation,0, decisions));
```
Comment (shoekzema) [inline speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backendUnitTests/SocketBlackBoxTests.h:41]:
```suggestion
	ASSERT_FALSE(bb.GetDecisions(&car, &situation, 0, decisions));
```

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backendUnitTests/SocketBlackBoxTests.h [hunk 2]
```diff
@@ -103,7 +75,7 @@ TEST(SocketBlackBoxTests, SocketTest)
 	// sends required and sending data of client
 	msgpack::sbuffer sbuffer;
 	msgpack::pack(sbuffer, order);
-	client.SendData(sbuffer.data(), sbuffer.size());
+	ASSERT_EQ(client.SendData(sbuffer.data(), sbuffer.size()),IPCLIB_SUCCEED);
```
Comment (shoekzema) [inline speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backendUnitTests/SocketBlackBoxTests.h:78]:
```suggestion
	ASSERT_EQ(client.SendData(sbuffer.data(), sbuffer.size()), IPCLIB_SUCCEED);
```

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backendUnitTests/SocketBlackBoxTests.h [hunk 3]
```diff
@@ -133,7 +106,7 @@ TEST(SocketBlackBoxTests, SocketTest)
 	};
 	sbuffer.clear();
 	msgpack::pack(sbuffer, action);
-	client.SendData(sbuffer.data(), sbuffer.size());
+	ASSERT_EQ(client.SendData(sbuffer.data(), sbuffer.size()),IPCLIB_SUCCEED);
```
Comment (shoekzema) [inline speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backendUnitTests/SocketBlackBoxTests.h:109]:
```suggestion
	ASSERT_EQ(client.SendData(sbuffer.data(), sbuffer.size()), IPCLIB_SUCCEED);
```

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backendUnitTests/SocketBlackBoxTests.h [hunk 4]
```diff
@@ -143,11 +116,11 @@ TEST(SocketBlackBoxTests, SocketTest)
 	msg3->convert(driveSituation2);
 
 	// test if the drivesituation is expected
-	DriveSituationMock exampleSituation = GetExampleDriveSituation();
+	BlackBoxDataMock exampleSituation = GetExampleBlackBoxDataMock();
 	TestDriveSituation(driveSituation2, exampleSituation);
 
 	// send back result of test 2
-	client.SendData(sbuffer.data(), sbuffer.size());
+	ASSERT_EQ(client.SendData(sbuffer.data(), sbuffer.size()),IPCLIB_SUCCEED);
```
Comment (shoekzema) [inline speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backendUnitTests/SocketBlackBoxTests.h:123]:
```suggestion
	ASSERT_EQ(client.SendData(sbuffer.data(), sbuffer.size()), IPCLIB_SUCCEED);
```

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backendUnitTests/SocketBlackBoxTests.h [hunk 5]
```diff
@@ -162,7 +135,7 @@ TEST(SocketBlackBoxTests, SocketTest)
 	TestDriveSituation(driveSituation3, mock);
 
 	// send back result of initial drive situation
-	client.SendData(sbuffer.data(), sbuffer.size());
+	ASSERT_EQ(client.SendData(sbuffer.data(), sbuffer.size()),IPCLIB_SUCCEED);
```
Comment (shoekzema) [inline speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backendUnitTests/SocketBlackBoxTests.h:138]:
```suggestion
	ASSERT_EQ(client.SendData(sbuffer.data(), sbuffer.size()), IPCLIB_SUCCEED);
```

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backendUnitTests/SocketBlackBoxTests.h [hunk 6]
```diff
@@ -174,14 +147,14 @@ TEST(SocketBlackBoxTests, SocketTest)
 
 	// tests if the drive situation is expected
 	TestDriveSituation(driveSituation4, exampleSituation);
-	client.SendData(sbuffer.data(), sbuffer.size());
+	ASSERT_EQ(client.SendData(sbuffer.data(), sbuffer.size()),IPCLIB_SUCCEED);
```
Comment (shoekzema) [inline speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backendUnitTests/SocketBlackBoxTests.h:150]:
```suggestion
	ASSERT_EQ(client.SendData(sbuffer.data(), sbuffer.size()), IPCLIB_SUCCEED);
```

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backendUnitTests/SocketBlackBoxTests.h [hunk 7]
```diff
@@ -174,14 +147,14 @@ TEST(SocketBlackBoxTests, SocketTest)
...
 
 	// gets a stop command
 	client.AwaitData(buffer, TEST_BUFFER_SIZE);
 	ASSERT_TRUE(buffer[0] == 'S' && buffer[1] == 'T' && buffer[2] == 'O' && buffer[3] == 'P');
 
 	// return to break connection
-	client.SendData("OK", 2);
+	ASSERT_EQ(client.SendData("OK", 2),IPCLIB_SUCCEED);
```
Comment (shoekzema) [inline speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backendUnitTests/SocketBlackBoxTests.h:157]:
```suggestion
	ASSERT_EQ(client.SendData("OK", 2), IPCLIB_SUCCEED);
```

**Review by shoekzema — CHANGES_REQUESTED** [review]

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/BlackBoxData.cpp [hunk 8]
```diff
@@ -0,0 +1,6 @@
+#include "BlackBoxData.h"
+
+BlackBoxData::BlackBoxData(tCarElt* p_car, tSituation* p_situation, int p_tickCount)
```
[the remark this answers is not in the document]
Comment (KungFuDonkey) [reply speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/BlackBoxData.cpp:3]: will be done in the other branch

**Review by shoekzema — APPROVED** [review]

**Review by noutheesink — APPROVED** [review]

**Pull request merged**